src/krate.rs

```diff
@@ -1100,6 +1123,7 @@ pub fn show(req: &mut Request) -> CargoResult<Response> {
 }
 
 /// Handles the `PUT /crates/new` route.
+/// Used by `cargo publish` to add a new crate or publish an existing crate.
```

Comment:
I would say "to publish a new crate or to publish a new version of an existing crate".